lua/nvim_lsp/util.lua

```diff
@@ -47,7 +47,7 @@ function M.tbl_deep_extend(dst, ...)
       if type(v) == 'table' and not vim.tbl_islist(v) then
         dst[k] = M.tbl_deep_extend(dst[k] or {}, v)
       else
-        dst[k] = v
+        dst[k] = dst[k] or v
```

Comment:
This doesn't seem like a universally correct choice. Maybe `tbl_deep_extend` needs a `behavior` parameter?

But the old behavior seems questionable too: I would not expect "extend" to overwrite each item in a list. 

Hope  @norcalli can clarify